src/reducers/notifications.ts

```diff
@@ -0,0 +1,52 @@
+import { NotifyClientTypes } from '@walletconnect/notify-client'
+
+export interface TopicNotificationsState {
+  fullNotifications: NotifyClientTypes.NotifyMessage[]
+  existingIds: Set<string>
+  hasMore: boolean
+}
+
+export interface NotificationsState {
+  [topic: string]: TopicNotificationsState
+}
+
+export type NotificationsActions = {
+  type: 'FETCH_NOTIFICATIONS'
+  notifications: NotifyClientTypes.NotifyMessage[]
+  topic: string
+  hasMore: boolean
+}
+
+// Opted for a reducer since the state changes are complex enough to warrant
+// changes to a set and an array and having all that inside the hooks would
```

Comment:
Readability. 

Fully agree on opting for a reducer pattern here btw, if anything generally underutilised these days with the over-emphasis on `useState + useEffect`

```suggestion
// changes to a set and an array. Having all that inside the hooks would
```